tests/core/test_axes.py

```diff
@@ -1,182 +1,149 @@
+    simulation_builder.add_person_entity(persons, {'Alicia': {}, 'Javier': {}, 'Tom': {}})
+    simulation_builder.add_group_entity('persons', ['Alicia', 'Javier', 'Tom'], group_entity, {
+        'housea': {'parents': ['Alicia']},
+        'houseb': {'parents': ['Tom'], 'children': ['Javier']},
+        })
+    simulation_builder.register_variable('salary', persons)
+    simulation_builder.add_parallel_axis({'count': 2, 'name': 'salary', 'min': 0, 'max': 3000, 'period': '2018-11'})
+    simulation_builder.expand_axes()
+    assert [role.key for role in simulation_builder.get_roles('households')] == ['parent', 'child', 'parent', 'parent', 'child', 'parent']
+
+
+def test_add_axis_distributes_memberships(simulation_builder, persons, group_entity):
+    simulation_builder.add_person_entity(persons, {'Alicia': {}, 'Javier': {}, 'Tom': {}})
+    simulation_builder.add_group_entity('persons', ['Alicia', 'Javier', 'Tom'], group_entity, {
+        'housea': {'parents': ['Alicia']},
+        'houseb': {'parents': ['Tom'], 'children': ['Javier']},
+        })
+    simulation_builder.register_variable('rent', group_entity)
+    simulation_builder.add_parallel_axis({'count': 2, 'name': 'rent', 'min': 0, 'max': 3000, 'period': '2018-11'})
+    simulation_builder.expand_axes()
+    assert simulation_builder.get_memberships('households') == [0, 1, 1, 2, 3, 3]
+
+
+def test_add_perpendicular_axes(simulation_builder, persons):
+    simulation_builder.add_person_entity(persons, {'Alicia': {}})
+    simulation_builder.register_variable('salary', persons)
+    simulation_builder.register_variable('pension', persons)
+    simulation_builder.add_parallel_axis({'count': 3, 'name': 'salary', 'min': 0, 'max': 3000, 'period': '2018-11'})
+    simulation_builder.add_perpendicular_axis({'count': 2, 'name': 'pension', 'min': 0, 'max': 2000, 'period': '2018-11'})
+    simulation_builder.expand_axes()
+    assert simulation_builder.get_input('salary', '2018-11') == approx([0, 1500, 3000, 0, 1500, 3000])
+    assert simulation_builder.get_input('pension', '2018-11') == approx([0, 0, 0, 2000, 2000, 2000])
+
+# Integration test
+
+
+def test_simulation_with_axes(simulation_builder):
+    from .test_countries import tax_benefit_system
+    input_yaml = """
+        persons:
+          Alicia: {salary: {2018-11: 0}}
+          Javier: {}
+          Tom: {}
+        households:
+          housea:
+            parents: [Alicia, Javier]
+          houseb:
+            parents: [Tom]
+        axes:
+            -
+                - count: 2
+                  name: rent
+                  min: 0
+                  max: 3000
+                  period: 2018-11
+    """
+    data = yaml.load(input_yaml)
+    simulation = simulation_builder.build_from_dict(tax_benefit_system, data)
+    assert simulation.get_array('salary', '2018-11') == approx([0, 0, 0, 0, 0, 0])
+    assert simulation.get_array('rent', '2018-11') == approx([0, 0, 3000, 0])
```

Comment:
Hey @Morendil , I don't understand why we expect a 0 as the last element of this array.